python/activator/middleware_interface.py

```diff
@@ -103,6 +105,48 @@ def get_central_butler(central_repo: str, instrument_class: str):
                   )
 
 
+def flush_local_repo(repo_dir: str, central_butler: Butler):
+    """Clean up a local repository abandoned by a different PP instance.
+
+    The function attempts to pass datasets to the local repo; as with
+    `MiddlewareInterface.export_outputs`, only datasets that would not collide
+    with other pods are handled at this time.
+
+    Parameters
+    ----------
+    repo_dir : `str`
+        An absolute path to a local repo that is *not* being managed by
+        TemporaryDirectory or similar.
+    central_butler : `lsst.daf.butler.Butler`
+        Butler repo to which to attempt to flush the local repo's contents.
+    """
+    try:
```

Comment:
Looks good for today.   This reads somewhat similar to `_export_subset` and I wonder if they can be refactored for better code reuse later.

Reply:
Not easily. `_export_subset` tries to make its query as restrictive as possible, both to minimize side effects and to keep things fast. Here we don't have the context to do that. 😞